Improve docs for making new Specialist Document types

ChrisBAshton opened the pull request:
See commits for details.

Trello: https://trello.com/c/ojsFViiX/2746-create-new-specialist-document-and-finder-product-safety-alerts-unsafe-products-and-recalls-3

ChrisBAshton commented on docs/create-a-new-specialist-document-type.md:
```diff
@@ -8,6 +8,15 @@ applications.
+2. Add any new field definitions to [this file](https://github.com/alphagov/govuk-content-schemas/blob/master/formats/shared/definitions/_specialist_document.jsonnet)
+3. Add examples [as instructed](https://github.com/alphagov/govuk-content-schemas/blob/master/docs/adding-a-new-schema.md#examples).
+   You can copy and paste from another specialist document format, only changing what is necessary (you can leave the body and headers unchanged).
+
```
```suggestion

When the PR is reviewed and its tests passing, it can be merged and deployed at this point.
```

ChrisBAshton commented on docs/create-a-new-specialist-document-type.md:
```diff
@@ -8,6 +8,16 @@ applications.
+3. Add examples [as instructed](https://github.com/alphagov/govuk-content-schemas/blob/master/docs/adding-a-new-schema.md#examples).
+   You can copy and paste from another specialist document format, only changing what is necessary (you can leave the body and headers unchanged).
+
+When the PR is reviewed and its tests passing, it can be merged and deployed at this point.
```
```suggestion
When the PR is reviewed and its tests passing, it can be merged and deployed at this point.

````

ChrisBAshton committed:
Add 'pre-production' note to the 'create new doc type' guide

This is the guide where it is most likely to be relevant, and it is
otherwise easily missed.

ChrisBAshton committed:
Rename from "Creating" to "Create"

Titles & headings should be action-oriented.

ChrisBAshton committed:
Improve docs for making new Specialist Document types

- Move schema creation up so that it is the first step. If we do the Specialist Publisher changes first, our tests will fail because the schema hasn't been added to govuk-content-schemas yet.
- Link to recent PR for govuk-content-schemas.
- Add instruction to edit allowed_document_types.yml
- Remove link to 'suggested-workflows.md', as that file no longer exists.
- Document a better starting point for creating the example.
- Add note about when to deploy content schemas
  Until now, the advice has been to deploy at the end. But for the specialist-publisher tests to pass, we need the content schemas to have been updated first. There's no harm in having the content-schemas change live before it is in use.
- As part of this, I moved content ID stuff into govuk-content-schemas, now that that is the first step.
- Remove subheadings from Search API section
  It's easier to process mentally by just presenting the entire list of things that need editing, in one place.
- Move permissions to the end, as it's more of a "NB" comment, and no actions need to be taken.
- Improve link text in Deploy section
- Clarifies that the email signup page only needs work if you want checkboxes.
- Links to Finder Frontend.

nicholsj commented on README.md:
```diff
@@ -4,7 +4,7 @@ Publishing App for Specialist Documents.
 This application does not store documents in a database of its own. Instead it uses Publishing API to store and retrieve documents. MongoDB is used for storing local user records.
 
-Formats can be [flagged](https://github.com/alphagov/specialist-publisher/blob/f8e93142dfad6f3971a73c923b01f2e7352bdb54/lib/documents/schemas/tax_tribunal_decisions.json#L64) as `pre-production`. `pre-production` documents are only publishable on development and integration.
+Formats can be flagged as `pre-production` (only publishable on development and integration). Read [Create a new specialist document format](./docs/create-a-new-specialist-documnet-type.md) for details.
```
typo in the URL `documnet`

ChrisBAshton replied:
Well spotted!

ChrisBAshton committed:
Note that the permission has to be created manually

Also fixes up the list markdown (unnecessary space prefixes).

ChrisBAshton committed:
Clarify why the reindex job is needed

Referencing our findings from
https://trello.com/c/n75DpJ4X/2707-investigate-options-to-stop-us-having-to-reindex-out-of-hours-when-changing-a-specialist-finder-timeboxed-to-1-to-2-days.

hannako reviewed (approved):
Looks good

I have a card to add instructions (related to whether or not there is a need to reindex) if we are amending an existing  doc type.
